### ISSUE TITLE
Multi-line statements support for Python code

### ISSUE THREAD

**Issue by GitHunter0** [rstudio/rstudio#9014]
It would be really great to have RStudio support of multi-line statements extended to Python code as well.

![image](https://user-images.githubusercontent.com/26294221/108933205-80b4c780-7629-11eb-9b18-87bfc8fcbae0.png)

It is a simple feature that saves a huge amount of repetitive manual work selecting code lines to run. However, no other Python IDE has it as far as I know, unfortunately. 

Although solely for R code at the present moment, only RStudio has that feature, so I hope you could make it work for Python too. 

Thank you

**Comment by ronblum** [comment]
@GitHunter0 Thank you for the suggestion! I'll mark it as an enhancement to review as we continue development of RStudio, and specifically in improving our support of Python.

**Comment by GitHunter0** [comment]
Hey @ronblum , thanks man!

**Comment by ronblum** [comment]
Still worth considering as we continue with Python improvements.

**Comment by ntluong95** [comment]
Any plan to bring this feature in the upcoming version of Rstudio @ronblum ?

**Comment by ronblum** [comment]
This isn't slated for an update yet. One way in which we prioritize what enhancements to include is by understanding how many people would like the feature. For that, we often look at the number of upvotes that an issue gets. So the more 👍 we get for the original post (currently at 4), the more likely we are to add it to our roadmap.

Triage: Something to consider for Cherry Blossom.

**Comment by ntluong95** [comment]
Something similar to consider in VSCode @ronblum https://github.com/microsoft/vscode-jupyter/issues/1471

**Comment by liyonaj573** [comment]
how possible this feature can come in the upcoming update of Rstudio @kevinushey @hadley

**Comment by jmcphers** [comment]
Another upvote from the community here: https://community.rstudio.com/t/cannot-get-python-chunks-to-run-at-once-rstudio-runs-line-by-line/157462/2

**Comment by ntluong95** [comment]
I am not sure how difficult to implement this feature on Rstudio, but at least maintaining the same experience when executing R code and Python code is needed.
Noted that Python users tend to write code by single lines instead of piping in R, but some specific cases like define a function, for loop, method chains in Pandas (wrapping between () ) can be prioritized to have the feature

**Comment by elemn** [comment]
I was the individual referenced in the above thread. Bumping because it would be really great to have this feature in RStudio as it would make a more uniform experience transitioning between the few languages. Haven't been able to find a Python IDE that does as much as RStudio so I think it would be advantageous for users used to the workflow, even if stuff like piping aren't really a part of Python.

**Comment by GitHunter0** [comment]
> Something similar to consider in VSCode @ronblum [microsoft/vscode-jupyter#1471](https://github.com/microsoft/vscode-jupyter/issues/1471)

Another one: https://github.com/microsoft/vscode-python/issues/18105

**Comment by ntluong95** [comment]
The vscode extension looks very promising. Hope Rstudio team can find the way to bring here @ronblum @jmcphers

**Comment by GitHunter0** [comment]
> The vscode extension looks very promising. Hope Rstudio team can find the way to bring here @ronblum @jmcphers

Just to register, the extension you are referring is this: https://github.com/kylebarron/vscode-jupyter-python

**Comment by GitHunter0** [comment]
> But it is for vscode? I meant I want to have the same in Rstudio

Yes, it is just for VScode, I'm only mentioning it as a reference to help build one for RStudio.
